lua/lvim/core/builtins/init.lua

```diff
@@ -23,12 +23,28 @@ local builtins = {
   "lvim.core.mason",
 }
 
+---@class LvimBuiltin
+---@field active boolean is builtin enabled
+---@field on_config_done function function called after the builtin is set up
+
 function M.config(config)
+  ---@type {[string]: LvimBuiltin}
+  lvim.builtin = {}
+
   for _, builtin_path in ipairs(builtins) do
     local builtin = reload(builtin_path)
 
     builtin.config(config)
   end
 end
 
+function M.setup_builtin(modname)
```

Comment:
no need to repeat redundant name

```suggestion
function M.setup(modname)
```